lib/lib.md

```diff
@@ -0,0 +1,106 @@
+# Module lib
+
+StarkNet.kt is a library allowing for easy interaction with StarkNet gateway and nodes, including
+querying starknet state, executing transactions and deploying contracts.
+
+Although written in Kotlin, StarkNet.kt has been created with compatibility with Java in mind.
+
+## Example usages in Java
+
+```java
+public class StarknetExample {
+    public static void main(String[] args) {
+        // Create a provider for interacting with StarkNet
+        var provider = new GatewayProvider("feeder_gateway_url", "gateway_url", StarknetChainId.TESTNET);
+        
+        // Create an account interface
+        Felt accountAddress = new Felt(45234235672347L);
+        Felt privateKey = new Felt(5326326273453L);
+        var account = new StandardAccount(provider, accountAddress, privateKey);
+        
+        // Make a request
+        Felt contractAddress = new Felt(42362362436L);
+        Felt storageKey = new Felt(13241253414L);
+        Request<GetStorageAtResponse> request = account.getStorageAt(contractAddress, storageKey, BlockTag.LATEST);
+
+        GetStorageAtResponse response = request.send();
+        response.getResult();
+    }
+}
+```
+
+## Example usages in Kotlin
+
+```kotlin
+fun main() {
+    // Create a provider for interacting with StarkNet
+    val provider = GatewayProvider("feeder_gateway_url", "gateway_url", StarknetChainId.TESTNET)
```

Comment:
Don't we provide those values as some constants? We probably should do that or expose provider configured for testnet already.

Reply:
I don't think we have these declared as constants at the moment.

Reply:
Then we should do something about it

Reply:
These has been added along with gateway client factory. Examples will need to be updated to reflect these changes.

Reply:
Where is the factory?

Reply:
In the GatewayProvider companion object

Reply:
Yes, please update the docs to use makeTestnetClient